### PR TITLE
Ruby group capstone - Catalog of my things 

### PR DESCRIPTION
**Pull request by keroloussamy**
## Add logic for:

> Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) have been created in a separate .rb file.

> All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes have been implemented.

> All methods visible in the diagram have been implemented:
- `add_item` method (in all classes that have that method)
- `can_be_archived?()` in the `Item` class
- `move_to_archive()` in the `Item` class
- `can_be_archived?()` in the `Book` class
- `can_be_archived?()` in the `MusicAlbum` class
- `can_be_archived?()` in the `Movie` class
- `can_be_archived?()` in the `Game` class

> Add unit tests for all implemented methods.

## User interface:

> Created a `main.rb` file that will serve as the console app entry-point.

> The console app, at the start, presents the user with a list of options to perform then let users choose an option and have a way to quit the app.

> All data have been preserved by saving collections in `.json` files.

## Database:

> Created a `schema.sql` file with tables that will be analogical to the structure of the classes in the app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- SQL Postgres
+
+
+## Getting Started
+To get a local copy up and running follow these simple example steps.
+
+- Click on the top right green "code" button.
+- On the dropdown menu, choose "download with zip" button.
+- After download, extract the zip file and you have the project on your machine.
+- Run `rubu main.rb` to run the application.
```
Comment (SajjadAhmad14) [inline README.md]: Please add the `bundle install` command in README so the user can install project dependencies.
Please correct the typo here as it should be `ruby main.rb`.
Please add instructions for the user to run the tests.

File: classes/item.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  attr_accessor :genre, :author, :label, :publish_date
+  attr_reader :id, :archived
+
+  def initialize(publish_date, archived, id: nil)
+    @id = id || Random.rand(1..1000)
+    @publish_date = publish_date
+    @archived = archived
+  end
+
+  def add_genre(genre)
```
Comment (SajjadAhmad14) [inline classes/item.rb:13]: Please add unit tests for all public methods of the `Item` class.

File: schema/schema.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+CREATE TABLE music_albums (
+  id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  genre_id FOREIGN KEY(id) REFERENCES genre(id),
+  author_id FOREIGN KEY(id) REFERENCES author(id),
+  label_id FOREIGN KEY(id) REFERENCES label(id),
+  publish_date DATE,
+  archived BOOLEAN,
+  on_spotify BOOLEAN
+);
```
Comment (SajjadAhmad14) [inline schema/schema.sql]: Please add indexes to all the foreign keys in the database. This is the basic solution to maximize the performance of queries.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and description. :heavy_check_mark:
All tests are passed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Hard work pays, keep up the good work 👍

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**